frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java

```diff
@@ -24,6 +31,20 @@
       @Nullable MacheteExecutionSettings settings,
       ExternalSystemTaskNotificationListener listener)
       throws ExternalSystemException, IllegalArgumentException, IllegalStateException {
+
+    var graphTableProvider = Option.of(settings)
+        .map(s -> s.getProject().getService(GraphTableProvider.class));
+    var hasSelectedGitMacheteTab = Option.of(settings)
+        .map(s -> s.getProject())
+        .map(this::hasSelectedGitMacheteTab)
+        .getOrElse(Boolean.FALSE);
```

Comment:
Or just:
```suggestion
    var hasSelectedGitMacheteTab = settings != null && hasSelectedGitMacheteTab(settings.getProject());
```

Reply:
Rule of the thumb: `.getOrElse(false/true)` can almost always be rewritten to sth more readable/less "wrapped"